docs/pages/router/reference/link-preview.mdx

```diff
@@ -0,0 +1,200 @@
+  src="/static/images/expo-router/link-preview-menu.png"
+  className="max-h-[400px]"
+/>
+
+### Icons
+
+You can specify an icon for each menu action using [SF Symbols](https://developer.apple.com/sf-symbols/).
+
+```tsx
+<Link href="/about">
+  <Link.Trigger>About</Link.Trigger>
+  <Link.Menu>
+    <Link.MenuAction title="Share" icon="square.and.arrow.up" onPress={handleSharePress} />
+    <Link.MenuAction title="Block" icon="nosign" onPress={handleBlockPress} />
+    <Link.MenuAction
+      title="Follow"
+      icon="person.crop.circle.badge.plus"
+      onPress={handleFollowPress}
+    />
+    <Link.MenuAction title="Copy" icon="doc.on.doc" onPress={handleCopyPress} />
+  </Link.Menu>
+</Link>
+```
+
+The following example shows a context menu with four elements, each using a different icon on iOS:
+
+<ContentSpotlight
+  alt="A screenshot of context menu with four elements."
+  src="/static/images/expo-router/link-preview-icons.png"
+  className="max-h-[400px]"
+/>
+
+### Nested menus
+
+You can nest menus by placing a `<Link.Menu>` inside another menu:
+
+```jsx
+<Link href="...">
+  <Link.Trigger>About</Link.Trigger>
+  <Link.Menu>
+    <Link.MenuAction title="Share" icon="square.and.arrow.up" onPress={() => {}} />
+    <Link.Menu title="More" icon="ellipsis">
+      <Link.MenuAction title="Copy" icon="doc.on.doc" onPress={() => {}} />
+      <Link.MenuAction title="Delete" icon="trash" destructive onPress={() => {}} />
+    </Link.Menu>
+  </Link.Menu>
+</Link>
+```
+
+The following example shows a nested context menu on iOS:
+
+<ContentSpotlight
+  alt="A screenshot of a nested context menu."
+  src="/static/images/expo-router/link-preview-nested-menu.png"
+  className="max-h-[400px]"
+/>
+
+### More customization options
+
+To explore all available customization options, see the API documentation for [`Link.MenuAction`](/versions/latest/sdk/router/#linkmenuaction).
+
+## Detecting if component is in preview
+
+If you're building a component that might be rendered inside a preview, you can use the `useIsPreview()` hook to adjust its behavior accordingly:
+
+```jsx
+function MyComponent() {
+  // This will be true if component/screen is being rendered inside a preview
+  const isInsidePreview = useIsPreview();
+
+  return isInsidePreview ? <Text>From within preview</Text> : <Text>I am outside of preview</Text>;
+}
+```
+
+## Known limitations
+
+### `replace` not Supported
+
+Using link previews with [`replace`](/versions/latest/sdk/router/#replace) mode is currently **not supported**. Previews can only be used with the default [`push`](/versions/latest/sdk/router/#push) navigation mode.
+
+### JavaScript tabs and slots
+
+When navigating within JavaScript tabs (rather than native tabs) or `<Slot>`s, preview transition animations may appear clunky. This is due to React's delayed rendering while the native preview animation begins immediately. To prevent this issues use, native tabs and stack navigators.
```

Comment:
```suggestion
When navigating within JavaScript tabs (rather than native tabs) or [`Slot`s](/versions/latest/sdk/router/#slot), preview transition animations may appear clunky. This is due to React's delayed rendering, while the native preview animation begins immediately. To prevent these issues, use native tabs and stack navigators.
```